http-client-core/src/main/java/io/micronaut/http/client/interceptor/HttpClientIntroductionAdvice.java

```diff
@@ -578,4 +691,24 @@ private String appendQuery(String uri, Map<String, List<String>> queryParams) {
         }
         return uri;
     }
+
+    private record RequestBinderResult(MutableHttpRequest<?> request,
+                                       Object errorResult,
+                                       boolean isError) {
+
+        private RequestBinderResult(@Nullable MutableHttpRequest<?> request, @Nullable Object errorResult, boolean isError) {
+            this.request = request;
+            this.errorResult = errorResult;
+            this.isError = isError;
+        }
+
+        static RequestBinderResult withRequest(@NonNull MutableHttpRequest<?> request) {
+            Objects.requireNonNull(request, "Bound HTTP request must not be null");
+            return new RequestBinderResult(request, null, false);
+        }
+
+        static RequestBinderResult withErrorResult(@Nullable Object errorResult) {
+            return new RequestBinderResult(null, errorResult, true);
+        }
+    }
```

Comment:
```suggestion
    private record RequestBinderResult(
        @Nullable MutableHttpRequest<?> request,
        @Nullable Object errorResult,
        boolean isError
    ) {

        static RequestBinderResult withRequest(@NonNull MutableHttpRequest<?> request) {
            Objects.requireNonNull(request, "Bound HTTP request must not be null");
            return new RequestBinderResult(request, null, false);
        }

        static RequestBinderResult withErrorResult(@Nullable Object errorResult) {
            return new RequestBinderResult(null, errorResult, true);
        }
    }
```

Constructor is implied in `record` I think... 🤔

Reply:
Yes, it is. Thanks. I missed that IntelliJ didn't remove it when refactored from a class.